From 96620d7773a516eea07fccbf2e666372670fa385 Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Wed, 20 Oct 2021 17:19:48 +0300
Subject: [PATCH 07/11] s32gen1: dts: change FIRC clock frequency

Previously, we used the 51MHz frequency which is the maximum frequency that
the FIRC clock may physically have. Bootrom uses this setting so that the
PLL clocks wouldn't output a clock beyond limits for some components (as
it would if it used the ideal 48MHz value).

Unfortunately, we also cannot use the FM CMU (frequency meter clock monitor
unit) because of the high tolerance margins for FIRC and FXOSC (external
oscillator). Therefore, we have FIRC as a fixed clock, set at 48MHz.

Note that, unlike bootrom, we provide FXOSC as a source for PLLs.

Issue: ALB-7938

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/fsl-s32-gen1.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/dts/fsl-s32-gen1.dtsi b/arch/arm/dts/fsl-s32-gen1.dtsi
index 9ff1faa931..764187acaa 100644
--- a/arch/arm/dts/fsl-s32-gen1.dtsi
+++ b/arch/arm/dts/fsl-s32-gen1.dtsi
@@ -62,7 +62,7 @@
 
 	firc: firc@0 {
 		compatible = "fixed-clock";
-		clock-frequency = <51000000>;
+		clock-frequency = <48000000>;
 		#clock-cells = <1>;
 	};
 
-- 
2.17.1

